docs/components/List/List.stories.mdx

```diff
@@ -1,9 +1,175 @@
+
+<Canvas>
+  <Card>
+    <List
+      items={[
+        {
+          id: 1,
+          content: "Content",
+          icon: "work",
+          iconColor: "interactive",
+          title: "Title",
+          value: "Value",
+          section: "Section 1",
+          onClick: () => alert("First list item clicked"),
+        },
+        {
+          id: 2,
+          content: "Content",
+          icon: "work",
+          iconColor: "interactive",
+          title: "Title",
+          value: "Value",
+          section: "Section 1",
+          onClick: () => alert("Second list item clicked"),
+        },
+        {
+          id: 3,
+          content: "Content",
+          icon: "work",
+          iconColor: "interactive",
+          title: "Title",
+          value: "Value",
+          section: "Section 2",
+          onClick: () => alert("Second list item clicked"),
+        },
+      ]}
+    />
+  </Card>
+</Canvas>
+
+### **Custom Rendering (Advanced Usage)**
+
+If you need more control over the rendering of list items, you can provide a
+**custom render function** using the `customRenderItem` prop. This is useful if
+the default rendering of the `ListItem` component doesn’t meet your
+requirements. If a custom render function is provided onClick, url, and hover
+functionality will still work. Sectioned list functionality will also remain the
+same.
+
+**Example: Custom Render Function with Custom Interface**
+
+```tsx
+import { List } from "./List";
+
+interface ProductListItemProps extends BaseListItemProps {
+  readonly name: string;
+  readonly price: string;
+  readonly section: string;
+  readonly onClick?: () => void;
+}
+
+const items: ProductListItemProps[] = [
+  { id: 1, name: "Lawn Mower", price: "$250.00", section: "Garden Tools", onClick: () => alert("Lawn Mower clicked") },
+  { id: 2, name: "Hedge Trimmer", price: "$85.00", section: "Garden Tools", onClick: () => alert("Hedge Trimmer clicked") },
+  { id: 3, name: "Drill", price: "$50.00", section: "Power Tools", onClick: () => alert("Drill clicked") },
+  { id: 4, name: "Screwdriver", price: "$10.00", section: "Hand Tools", onClick: () => alert("Screwdriver clicked") },
+];
+
+
+const renderProductItem = (item: ProductListItemProps) => (
+    <Flex template={["shrink", "grow"]} align="start">
+        <Text>{item.price}</Text>
+        <Heading level={4}>{item.name}</Heading>
+    </Flex>
+  </div>
+);
+
+export const CustomRenderExample = () => (
+  <List items={items} customRenderItem={renderProductItem} />
+);
+
+```
+
+**In this example**:
+
+- We define a custom interface (`ProductListItemProps`) that extends
+  `BaseListItemProps`.
+- The `customRenderItem` function provides complete control over how each item
+  is rendered.
+
+**When to Use Custom Rendering:**
+
+- Use a custom render function when you need complex layouts that the `ListItem`
+  component can’t provide.
+- Create your own interface extending `BaseListItemProps` to define the fields
```

Comment:
nitpick: this is less about "when to use custom rendering" and more about how

maybe a heading of `About Custom Rendering` would encapsulate both of these points. the first describing "when" to use it, and the second "how"